pkg/controller/garbagecollector/garbage_collector_utils.go

```diff
@@ -0,0 +1,37 @@
+package garbagecollector
+
+import (
+	"strings"
+
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/typed/discovery"
+	utilerrors "k8s.io/kubernetes/pkg/util/errors"
+)
+
+// TODO: put this function to the discovery package.
+func ServerPreferredGroupVersionResources(discoveryClient discovery.DiscoveryInterface) ([]unversioned.GroupVersionResource, error) {
```

Comment:
Can this be a method of the discovery client?

Please explain why one might call it in the documentation line.